python/src/main/python/yaml-template/main.py

```diff
@@ -69,74 +60,12 @@ def _preparse_jinja_flags(argv):
   return pipeline_args
 
 
-def _configure_parser(argv):
-  parser = argparse.ArgumentParser()
-  parser.add_argument(
-    '--yaml_pipeline',
-    '--pipeline_spec',
-    help='A yaml description of the pipeline to run.')
-  parser.add_argument(
-    '--yaml_pipeline_file',
-    '--pipeline_spec_file',
-    help='A file containing a yaml description of the pipeline to run.')
-  parser.add_argument(
-    '--json_schema_validation',
-    default='generic',
-    help='none: do no pipeline validation against the schema; '
-         'generic: validate the pipeline shape, but not individual transforms; '
-         'per_transform: also validate the config of known transforms')
-  parser.add_argument(
-    '--jinja_variables',
-    default=None,
-    type=json.loads,
-    help='A json dict of variables used when invoking the jinja preprocessor '
-         'on the provided yaml pipeline.')
-  return parser.parse_known_args(argv)
-
-
-def _pipeline_spec_from_args(known_args):
-  if known_args.yaml_pipeline_file and known_args.yaml_pipeline:
-    raise ValueError(
-      "Exactly one of yaml_pipeline or yaml_pipeline_file must be set.")
-  elif known_args.yaml_pipeline_file:
-    with FileSystems.open(known_args.yaml_pipeline_file) as fin:
-      pipeline_yaml = fin.read().decode()
-  elif known_args.yaml_pipeline:
-    pipeline_yaml = known_args.yaml_pipeline
-  else:
-    raise ValueError(
-      "Exactly one of yaml_pipeline or yaml_pipeline_file must be set.")
-
-  return pipeline_yaml
-
-
-class _BeamFileIOLoader(jinja2.BaseLoader):
-  def get_source(self, environment, path):
-    source = FileSystems.open(path).read().decode()
-    return source, path, lambda: True
-
-
 def run(argv=None):
-  argv = _preparse_jinja_flags(argv)
-  known_args, pipeline_args = _configure_parser(argv)
-  pipeline_yaml = (  # keep formatting
-    jinja2.Environment(
-      undefined=jinja2.StrictUndefined, loader=_BeamFileIOLoader())
-    .from_string(_pipeline_spec_from_args(known_args))
-    .render(**known_args.jinja_variables or {}))
-  pipeline_spec = yaml.load(pipeline_yaml, Loader=yaml_transform.SafeLineLoader)
-
-  with beam.Pipeline(  # linebreak for better yapf formatting
-          options=beam.options.pipeline_options.PipelineOptions(
-            pipeline_args,
-            pickle_library='cloudpickle',
-            **yaml_transform.SafeLineLoader.strip_metadata(pipeline_spec.get(
-              'options', {}))),
-          display_data={'yaml': pipeline_yaml}) as p:
-    print("Building pipeline...")
-    yaml_transform.expand_pipeline(
-      p, pipeline_spec, validate_schema=known_args.json_schema_validation)
-    print("Running pipeline...")
+  parser = argparse.ArgumentParser()
+  _, pipeline_args = parser.parse_known_args(_preparse_jinja_flags(argv))
+  pipeline_args += ['--sdk_location=container']
```

Comment:
Why do we want this flag? Was it just for testing the dev release?

Reply:
This should help with container startup speed according to docs

https://cloud.google.com/dataflow/docs/guides/build-container-image#pre-build_a_container_image_when_submitting_the_job

"This option prevents the workers from downloading the SDK when your job launches. Instead, the SDK is retrieved directly from the container image."

Reply:
We should look at the code, but I'm pretty sure this doesn't apply to released versions that already assume the SDK is installed in the container.

Reply:
Just ran a speed test and it ran in about the same time with and without the flag specified, so I will remove